RecoHI/HiJetAlgos/plugins/VoronoiBackgroundProducer.cc

```diff
@@ -143,7 +162,12 @@ VoronoiBackgroundProducer::produce(edm::Event& iEvent, const edm::EventSetup& iS
 	 ue = new UECalibration(ueTable->values);
 	}
 
-     voronoi_ = new VoronoiAlgorithm(ue,equalizeR_,std::pair<double, double>(equalizeThreshold0_,equalizeThreshold1_),doEqualize_);
+	if (sameEvent_) {
+		voronoi_ = new SameEventVoronoiAlgorithm(sameEventAntiktDistance_, sameEventExclusionPtMin_, sameEventExclusionRadius_, sameEventFakeRejectEtMax_, sameEventFakeRejectEtMaxOverMean_, std::pair<double, double>(equalizeThreshold0_,equalizeThreshold1_), doEqualize_);
+	}
+	else {
+	voronoi_ = new VoronoiAlgorithm(ue,equalizeR_,excludeV1_,maxVn_,diagonalVn_,std::pair<double, double>(equalizeThreshold0_,equalizeThreshold1_),doEqualize_);
+	}
```

Comment:
Is voronoi_ ever deleted? It might be a good idea to convert it into a unique_ptr to protect against memory leaks.